embrace-android-delivery/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/execution/RequestExecutionServiceImpl.kt

```diff
@@ -1,15 +1,86 @@
 package io.embrace.android.embracesdk.internal.delivery.execution
 
+import io.embrace.android.embracesdk.internal.comms.api.ApiRequestV2
 import io.embrace.android.embracesdk.internal.comms.api.ApiResponse
+import io.embrace.android.embracesdk.internal.comms.api.Endpoint
 import io.embrace.android.embracesdk.internal.delivery.SupportedEnvelopeType
+import okhttp3.Headers.Companion.toHeaders
+import okhttp3.OkHttpClient
+import okhttp3.Protocol
+import okhttp3.Request
+import okhttp3.RequestBody
+import okio.BufferedSink
+import okio.buffer
+import okio.source
 import java.io.InputStream
 
-class RequestExecutionServiceImpl : RequestExecutionService {
+class RequestExecutionServiceImpl(
+    private val coreBaseUrl: String,
+    private val lazyDeviceId: Lazy<String>,
+    private val appId: String,
+    private val embraceVersionName: String,
+) : RequestExecutionService {
+
+    private val okHttpClient = OkHttpClient()
+        .newBuilder()
+        .protocols(listOf(Protocol.H2_PRIOR_KNOWLEDGE))
+        .build()
 
     override fun attemptHttpRequest(
         payloadStream: () -> InputStream,
-        envelopeType: SupportedEnvelopeType
+        envelopeType: SupportedEnvelopeType,
     ): ApiResponse {
-        return ApiResponse.None
+        val apiRequest = envelopeType.endpoint.getApiRequestFromEndpoint()
+
+        val requestBody = object : RequestBody() {
+            override fun contentType() = null
+
+            override fun writeTo(sink: BufferedSink) {
+                payloadStream().source().buffer().use { source ->
+                    sink.writeAll(source)
+                }
+            }
+        }
+
+        val request = Request.Builder()
+            .url(apiRequest.url)
+            .headers(apiRequest.getHeaders().toHeaders())
+            .post(requestBody)
+            .build()
+
+        okHttpClient.newCall(request).execute().use { response ->
+            if (response.isSuccessful) {
```

Comment:
It will be necessary to support the same responses that `ApiClientImpl` currently does